c_src/sqlite3_nif.c

```diff
@@ -448,194 +448,124 @@ exqlite_prepare(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
-                    if (rc == SQLITE_OK) {
-                        return make_atom(env, "ok");
-                    }
-
-                    return enif_raise_exception(
-                      env,
-                      make_bind_error(
-                        env,
-                        make_message(env, "Failed to bind argument as blob"),
-                        arg));
-                }
-            }
-        }
-    }
+    ErlNifBinary blob;
+    if (!enif_inspect_binary(env, argv[2], &blob))
+        return raise_badarg(env, argv[2]);
 
-    return enif_raise_exception(
-      env,
-      make_bind_error(
-        env,
-        make_message(env, "Failed to bind argument"),
-        arg));
+    int rc = sqlite3_bind_blob(statement->statement, idx, (char*)blob.data, blob.size, SQLITE_TRANSIENT);
+    return enif_make_int(env, rc);
 }
 
-///
-/// @brief Binds arguments to the sql statement
-///
 static ERL_NIF_TERM
-exqlite_bind(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
+exqlite_bind_integer(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
 {
-    assert(env);
+    assert(argc == 3);
 
-    unsigned int parameter_count      = 0;
-    unsigned int argument_list_length = 0;
-    connection_t* conn                = NULL;
-    statement_t* statement            = NULL;
-    ERL_NIF_TERM list;
-    ERL_NIF_TERM head;
-    ERL_NIF_TERM tail;
+    statement_t* statement;
+    if (!enif_get_resource(env, argv[0], statement_type, (void**)&statement))
+        return raise_badarg(env, argv[0]);
 
-    if (argc != 3) {
-        return enif_make_badarg(env);
-    }
+    unsigned int idx;
+    if (!enif_get_uint(env, argv[1], &idx))
+        return raise_badarg(env, argv[1]);
 
-    if (!enif_get_resource(env, argv[0], connection_type, (void**)&conn)) {
-        return make_error_tuple(env, "invalid_connection");
-    }
+    ErlNifSInt64 i;
+    if (!enif_get_int64(env, argv[2], &i))
+        return raise_badarg(env, argv[2]);
 
-    if (!enif_get_resource(env, argv[1], statement_type, (void**)&statement)) {
-        return make_error_tuple(env, "invalid_statement");
-    }
+    int rc = sqlite3_bind_int64(statement->statement, idx, i);
+    return enif_make_int(env, rc);
+}
 
-    if (!enif_get_list_length(env, argv[2], &argument_list_length)) {
-        return make_error_tuple(env, "bad_argument_list");
-    }
+static ERL_NIF_TERM
+exqlite_bind_float(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
+{
+    assert(argc == 3);
 
-    parameter_count = (unsigned int)sqlite3_bind_parameter_count(statement->statement);
-    if (parameter_count != argument_list_length) {
-        return make_error_tuple(env, "arguments_wrong_length");
-    }
+    statement_t* statement;
+    if (!enif_get_resource(env, argv[0], statement_type, (void**)&statement))
+        return raise_badarg(env, argv[0]);
 
-    sqlite3_reset(statement->statement);
```

Comment:
Forgot about this one.

Reply:
https://github.com/elixir-sqlite/exqlite/pull/298/commits/c384bd17f6613aaf1f7a45dba5432a84347e099a